Modules/File/classes/Implementation/class.ilObjFileImplementationLegacy.php

```diff
@@ -279,6 +279,32 @@ public static function parseInfoParams($entry) : array
             $data[2] = "1";
         }
 
+        // BEGIN bugfix #31730
+        // if more than 2 commas are detected, the need for reassembling the filename is: possible to necessary
+        if (sizeof($data) > 2)
+        {
+          $last = sizeof($data) - 1;
+          for ($n = 1; $n < $last - 1; $n++)
+          {
+            $data[0] .= "," . $data[$n];
+          }
+
+          // trying to distinguish the next-to-last being a 'last part of the filename'
+          // or a 'version information',  based on having a dot included or not
+          if (str_contains($data[$last - 1], "."))
```

Comment:
Just a note: This is a PHP 8 function, which does not exist in the PHP 7.4 core. I know, this might work, but just because of `composer` (which define this function if it does not exists).

Reply:
check,  nice pifall for a PHP wannabe like me..  ;-)